File: backend/Models/Game.cs
namespace game_store.Models;

public class Game
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string Description { get; set; }
    public int GenreId { get; set; }
    public Genre? Genre { get; set; }
    public required DateOnly ReleaseDate { get; set; }
    public required int CompanyId { get; set; }
    public Company? Company { get; set; }
    public decimal Price { get; set; }
    public bool IsAvailable { get; set; }
    public string? CoverImageUrl { get; set; }
    public string? CardImageUrl { get; set; }

    public required List<Platform> Platforms { get; set; }

}
public enum Platform
{
    PC,
    PS5,
    XboxSeriesX,
    NintendoSwitch
}

File: frontend/src/types/game.ts
export type Platform = 'PC' | 'PS5' | 'XboxSeriesX' | 'NintendoSwitch';

export interface GetGamesDto {
  id: number;
  name: string;
  description: string;
  genreId: number;
  genreName: string;
  releaseDate: string;
  companyId: number;
  companyName: string;
  price: number;
  isAvailable: boolean;
  coverImageUrl: string | null;
  cardImageUrl: string | null;
  platforms: Platform[];
}

export interface Game {
  id: number;
  name: string;
  description: string;
  genreId: number;
  genre: Genre | null;
  releaseDate: string;
  companyId: number;
  company: Company | null;
  price: number;
  isAvailable: boolean;
  coverImageUrl: string | null;
  platforms: Platform[];
}

export interface Genre {
  id: number;
  name: string;
}

export interface Company {
  id: number;
  name: string;
}

export interface CartItem {
  game: GetGamesDto;
  platform: Platform;
}

export const platformIcons: Record<Platform, string> = {
  PC: '🖥',
  PS5: '🎮',
  XboxSeriesX: '🎯',
  NintendoSwitch: '🕹',
};

export const platformColors: Record<Platform, string> = {
  PC: '#00d4ff',
  PS5: '#003087',
  XboxSeriesX: '#107c10',
  NintendoSwitch: '#e60012',
};
